preprocess/cache_main.cc

```diff
@@ -50,14 +60,65 @@ struct QueueEntry {
   StringPiece *value;
 };
 
-void Input(util::UnboundedSingleQueue<QueueEntry> &queue, util::scoped_fd &process_input, std::unordered_map<uint64_t, StringPiece> &cache, std::size_t flush_rate) {
+void Input(util::UnboundedSingleQueue<QueueEntry> &queue, util::scoped_fd &process_input, std::unordered_map<uint64_t, StringPiece> &cache, std::size_t flush_rate, Options &options) {
   QueueEntry q_entry;
   {
     util::FakeOFStream process(process_input.get());
     std::pair<uint64_t, StringPiece> entry;
     std::size_t flush_count = flush_rate;
+    // Parse column numbers, if given using --key option, into an integer vector (comma separated integers)
+    std::vector<int> columns_args;
+
+    size_t pos_args = 0;
+    std::string token_args;
+    std::string s_args = options.key;
+    while ((pos_args = s_args.find(",")) != std::string::npos) {
+      token_args = s_args.substr(0, pos_args);
+      std::string::size_type sz;
+      int int_token_args = std::stoi (token_args, &sz);
+      columns_args.push_back(int_token_args);
+      s_args.erase(0, pos_args + 1);
+    }
+    std::string::size_type sz;
+    int last_int_token_args = std::stoi (s_args, &sz);
+    columns_args.push_back(last_int_token_args);
+
```

Comment:
See `ParseFields` in `fields.hh`.  There's already a parser for this and it supports `-`.